CaesarCipher/CaesarCipher/main.m

```diff
@@ -10,42 +10,287 @@
 
 @interface CaesarCipher : NSObject
 
-- (NSString *)decode:(NSString *)string offset:(int)offset;
-- (NSString *)encode:(NSString *)string offset:(int)offset;
+- (NSString *)decode:(NSString *)string offset:(NSInteger)offset;
+- (NSString *)encode:(NSString *)string offset:(NSInteger)offset;
+
+- (BOOL)easyIsSameMessage:(NSString *)msg1 asMessage:(NSString *)msg2;
+- (BOOL)hardIsSameMessage:(NSString *)msg1 asMessage:(NSString *)msg2;
 
 @end
 
-@implementation CaesarCipher
+@implementation CaesarCipher {
+    NSInteger _alphabetSize;
+}
+
+-(id)init {
+    if (self = [super init]) {
+        _alphabetSize = 26;
+    }
+    return self;
+}
 
-- (NSString *)encode:(NSString *)string offset:(int)offset {
-    if (offset > 25) {
-        NSAssert(offset < 26, @"offset is out of range. 1 - 25");
+- (NSString *)encode:(NSString *)string offset:(NSInteger)offset {
+    
+    // if offset is an illegal value, the assert statement will
+    // fail and the app will crash with the provided error message
+    if (offset > _alphabetSize - 1) {
+        NSAssert(offset < _alphabetSize, @"offset is out of range. 1 - 25");
```

Comment:
Yes, I had a suspicion that it should be a constant but I didn't really
know how to implement it properly. This was very helpful, thanks!

On Mon, Jun 29, 2015 at 12:24 PM, Mike Kavouras <notifications@github.com>
wrote:

> In CaesarCipher/CaesarCipher/main.m
> <https://github.com/accesscode-2-2/unit-0-hw-week-2/pull/25#discussion_r33482131>
> :
>
> >
> > -- (NSString *)encode:(NSString *)string offset:(int)offset {
> > -    if (offset > 25) {
> > -        NSAssert(offset < 26, @"offset is out of range. 1 - 25");
> > +- (NSString *)encode:(NSString *)string offset:(NSInteger)offset {
> > +
> > +    // if offset is an illegal value, the assert statement will
> > +    // fail and the app will crash with the provided error message
> > +    if (offset > _alphabetSize - 1) {
> > +        NSAssert(offset < _alphabetSize, @"offset is out of range. 1 - 25");
>
> Suggestion: If you're going to store _alphabetSize in a variable I
> wouldn't do it as a property. It's not a defining characteristic of the
> Cipher itself, it's just a piece of information that the Cipher will use to
> execute correctly. My suggestion is to make a *constant* at the top of
> your file. You can do this with the #define keyword.
>
> #define ALPHABET_SIZE 26
>
> By convention, constance are all caps. They also go at the top of your
> file.
>
> #import <Foundation/Foundation.h>
>
> #define ALPHABET_SIZE 26
> @interface CaesarCipher : NSObject
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/accesscode-2-2/unit-0-hw-week-2/pull/25/files#r33482131>
> .
>